src/_guides/language/effective-dart/design.md

```diff
@@ -1103,54 +1093,49 @@ or a `File` object? Is this method synchronous or asynchronous? This is clearer:
 
 {:.good-style}
 <?code-excerpt "misc/lib/effective_dart/design_good.dart (inferred)"?>
 {% prettify dart %}
 const screenWidth = 640; // Inferred as int.
 {% endprettify %}
 
-You may still wish to explicitly annotate even when inference would fill in the
-same type. If the inference relies on values or declarations from other
-libraries, you may want to annotate the type in *your* declaration so that a
-change to that other library doesn't silently change the type of your own API
-without you realizing.
+"Obvious" isn't precisely defined, but these are all good candidates:
 
-This guideline does *not* mean you can never use the dynamic type in a public
-signature, just that you should *annotate* it explicitly when you do so.
+* Literals.
+* Constructor invocations.
+* References to other constants that are explicitly typed.
+* Simple expressions on numbers and strings.
+* Factory methods like `int.parse()`, `Future.wait()`, etc. that readers are
+  expected to be familiar with.
 
+In general, err on the side of annotating. You may still wish to explicitly
+annotate even the type is obvious. If the inferred type relies on values or
+declarations from other libraries, you may want to annotate the type in *your*
+declaration so that a change to that other library doesn't silently change the
+type of your own API without you realizing.
 
-### PREFER type annotating private declarations whose type isn't inferred.
 
-Type annotations on your public declarations help *users* of your code. Nearly
-as important is guiding *maintainers* of your code. Adding type annotations to
-private member and variable declarations can help future readers of your code
-understand it, and help corral bugs.
+### CONSIDER type annotating private top-level variables and fields if the type isn't obvious.
 
-{:.good-style}
-<?code-excerpt "misc/lib/effective_dart/design_good.dart (type-private)"?>
-{% prettify dart %}
-class CallChainVisitor {
-  final SourceVisitor _visitor;
-  final Expression _target;
-  // ···
-  void _writeCall(Expression call) { ... }
-}
-{% endprettify %}
+Type annotations on your public declarations help *users* of your code. Types on
+private members help *maintainers*. The scope of a private declaration is
+smaller and those who need to know the type of that declaration are also more
+likely to be familiar with the surrounding code. That makes it reasonable to
+lean more heavily on inference and omit types in private declarations, which is
+why this guideline is softer than the previous one.
 
-This guideline is deliberately softer than the previous one because the benefits
-of static typing are smaller when code is confined within a single library. In
-practice, though, most Dart users type annotate all declarations, public or
-private.
+If you think the initializer expression&mdash;whatever it is&mdash;is
+sufficiently clear, then you may omit the annotation. But if you think
```

Comment:
Maybe:

> If you think the initializer expression&mdash;whatever it is&mdash;is sufficiently clear, then you may omit the annotation.

->

> If you think that an initializer expression type is sufficiently clear, then omit the annotation.

Reply:
Hmm, I think I like the original more. I don't think of expressions as "having" a type, and I wanted to draw attention to the fact that, unlike the previous rule, it's OK to infer a type even for non-"obvious" initializers.

Reply:
Ok